src/lib/rate_limiter_transport.go

```diff
@@ -33,19 +33,30 @@ func (t limitTransport) RoundTrip(req *http.Request) (*http.Response, error) {
 	return t.RoundTripper.RoundTrip(req)
 }
 
-var limiterOnce sync.Once
-var limiter ratelimit.Limiter
+type keyOnce struct {
+	m sync.Map // map[key]interface{}
+}
+
+func (ko *keyOnce) Do(key interface{}, fn func()) {
+	once := &sync.Once{}
+	actual, _ := ko.m.LoadOrStore(key, once)
+	actual.(*sync.Once).Do(fn)
+}
+
+var limiterOnce keyOnce
+var limiters sync.Map
 
-func newLimiter(rate int) ratelimit.Limiter {
-	limiterOnce.Do(func() {
-		limiter = ratelimit.New(rate)
+func newLimiter(key string, rate int) ratelimit.Limiter {
+	limiterOnce.Do(key, func() {
+		limiters.Store(key, ratelimit.New(rate))
 	})
-	return limiter
+	val, _ := limiters.Load(key)
+	return val.(ratelimit.Limiter)
 }
 
-func newRateLimitedTransport(rate int, transport http.RoundTripper) http.RoundTripper {
+func NewRateLimitedTransport(key string, rate int, transport http.RoundTripper) http.RoundTripper {
```

Comment:
I think we can simplify this part as this is a common library, we should not do the singleton in the common package, if we really need it we should do it in the business code.

Reply:
make sense